src/mcl_3dl.cpp

```diff
@@ -727,9 +727,9 @@ class MCL3dlNode
 
     if (!global_localization_fix_cnt_)
     {
-      if (std::sqrt(cov[0][0] + cov[1][1]) > params_.std_warn_thresh_[0] ||
-          std::sqrt(cov[2][2]) > params_.std_warn_thresh_[1] ||
-          std::sqrt(cov[5][5]) > params_.std_warn_thresh_[2])
+      if (std::sqrt(pose.pose.covariance[0] + pose.pose.covariance[7]) > params_.std_warn_thresh_[0] ||
+          std::sqrt(pose.pose.covariance[14]) > params_.std_warn_thresh_[1] ||
+          std::sqrt(pose.pose.covariance[35]) > params_.std_warn_thresh_[2])
```

Comment:
Maybe better to have a helper to convert [j, k] to flattened index or just write like `1 * 6 + 1`

Reply:
Done in f3a12d71b0b0b92c8553515a6771acd84a292d8c.